File: python/tournament_winner.py
def tournament_winner (competitions, results):
    points = {}
    leader = None
    higher_score = 0
    i = 0
    while i < len(competitions):
		# get actual winner
        winner = None
        if results[i] == 1:
            winner = competitions[i][0]
        else:
            winner = competitions[i][1]   
        # init or update
	    actual_points = 3        
        if winner in points:
            actual_points = points[winner] + 3        
        #update mapped values
	    points[winner] = actual_points
        # check who is the leader 
	    if actual_points > higher_score:
            higher_score = actual_points
            leader = winner
            i = i + 1
    return leader

File: javascript/tournament_winner.js
module.exports = class TournamentWinner {
    get(competitions, results) {
        const points = {};
        let leader = null;
        higherScore = 0;
        for (i = 0; i < competitions.length; i++) {
            const winner =
                results[i] === 1 ? competitions[i][0] : competitions[i][1];
            actualPoints = points[winner];
            actualPoints = !actualPoints ? 3 : actualPoints + 3;
            points[winner] = actualPoints;
            if (actualPoints > higherScore) {
                higherScore = actualPoints;
                leader = winner;
            }
        }
        return leader;
    }
};
